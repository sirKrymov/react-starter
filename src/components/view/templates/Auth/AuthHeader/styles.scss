.auth-header {
  position: absolute;
  padding: 0 24px;
  height: 60px;
  width: 100%;
  left: 0;
  top: 0;

  &__content {
    justify-content: space-between;
    align-items: center;
    display: flex;
    height: 100%;
  }
}
