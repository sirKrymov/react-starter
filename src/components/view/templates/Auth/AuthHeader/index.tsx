// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React from 'react';

import { useTranslation } from 'react-i18next';
import { observer } from 'mobx-react-lite';

import { ContentLimiter } from 'components/view/layouts/ContentLimiter';
import { LngChanger } from 'components/view/compound/LngChanger';
import { LinkButton } from 'components/view/atoms/LinkButton';
import { Switch } from 'components/view/atoms/Switch';
import { Logo } from 'components/view/atoms/Logo';

import { useStores } from 'hooks/useStores';
import { useToast } from 'hooks/useToasts';

import urls from 'constants/urls';

import './styles.scss';

interface IProps {
  page?: 'signin' | 'signup' | 'resetPassword' | 'newPassword' | 'verify-email';
}

export const AuthHeader = observer(({ page }: IProps) => {
  const { themeStore } = useStores();
  const { t } = useTranslation();
  const toast = useToast();

  const { setTheme, theme } = themeStore;

  function handleChange(): void {
    setTheme(theme === 'light' ? 'dark' : 'light');
  }

  return (
    <header className="auth-header">
      <ContentLimiter>
        <div className="auth-header__content">
          <Logo id="auth-logo" />

          <div style={{ display: 'flex', alignItems: 'center' }}>
            <button
              onClick={(): void =>
                toast.success('Message was sent successfully')
              }
            >
              Toasts
            </button>

            <LngChanger />

            <div
              style={{
                alignItems: 'center',
                margin: '0 16px',
                display: 'flex'
              }}
            >
              <Switch
                onChange={handleChange}
                value={theme === 'dark' ? true : false}
                name="darkModeSwitcher"
                id="dark-mode-switcher"
              />
            </div>

            {page === 'signin' && (
              <LinkButton
                variant="outlined"
                height="sm"
                width="content"
                to={urls.auth.signup}
                id="auth-sign-up-link"
              >
                {t('authHeader.signUpButtonTitle')}
              </LinkButton>
            )}
          </div>
        </div>
      </ContentLimiter>
    </header>
  );
});
