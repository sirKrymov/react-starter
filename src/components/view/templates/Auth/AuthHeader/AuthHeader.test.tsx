import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { AuthHeader } from '../AuthHeader';

describe('<AuthHeader> template sub component', () => {
  let shallowComponent: ShallowWrapper;

  const createComponent = (): void => {
    shallowComponent = shallow(<AuthHeader />);
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
