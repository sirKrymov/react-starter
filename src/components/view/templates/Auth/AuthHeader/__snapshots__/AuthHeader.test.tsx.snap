// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthHeader> template sub component WHEN the component loads THEN it should match snapshot 1`] = `
<header
  className="auth-header"
>
  <ContentLimiter>
    <div
      className="auth-header__content"
    >
      <Memo(LogoBase)
        id="auth-logo"
      />
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <button
          onClick={[Function]}
        >
          Toasts
        </button>
        <Memo(LngChangerBase) />
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "margin": "0 16px",
            }
          }
        >
          <Switch
            id="dark-mode-switcher"
            name="darkModeSwitcher"
            onChange={[Function]}
            value={false}
          />
        </div>
      </div>
    </div>
  </ContentLimiter>
</header>
`;
